### ISSUE TITLE
text_ansi not reporting correct size with WINDOW_ALWAYS_AUTO_RESIZE

### ISSUE THREAD

**Issue by learn-more** [swistakm/pyimgui#204]
Code to reproduce:
```
        def test_fn1():
            imgui.begin('Test1', flags=imgui.WINDOW_ALWAYS_AUTO_RESIZE)
            imgui.text_ansi('TEST\033[31m TEXT1')
            imgui.end()
        def test_fn2():
            imgui.begin('Test2', flags=imgui.WINDOW_ALWAYS_AUTO_RESIZE)
            imgui.text('TEST TEXT2')
            imgui.end()
```

Result:
![image](https://user-images.githubusercontent.com/610685/105635001-17426d00-5e61-11eb-9487-ed026708ee55.png)

**Comment by swistakm** [comment]
Yeah, I knew those ansi sequence coloring will be problematic. I suppose Imgui simply takes the length of whole string including escape characters.

**Comment by learn-more** [comment]
> Yeah, I knew those ansi sequence coloring will be problematic. I suppose Imgui simply takes the length of whole string including escape characters.

I looked briefly at this, and this were my notes:
- One option is to move the first few parsing loops from `ImFont_RenderAnsiText` to a new `CalcTextSizeAnsi`, that does the first pass of filling the global static arrays, but some state inbetween would have to be duplicated.
- Another option is to Parse the entire text twice, one while rendering it, and one while calculating the text size

Another (unrelated) thing I noticed:
`char_buf` in `AnsiTextColored.cpp` is updated but never used.
